nexus/src/app/sagas/instance_create.rs

```diff
@@ -187,9 +187,12 @@ impl NexusSaga for SagaInstanceCreate {
             Ok(())
         }
 
-        for (i, group) in
-            params.create_params.anti_affinity_groups.iter().enumerate()
+        let aa_groups = match params.create_params.anti_affinity_groups.as_ref()
```

Comment:
We also do the same thing here, because we're passing the same user-supplied instance creation params into this saga.